student_work/Kaloyan/PracticumWeek4/Computer/Computer.cpp

```diff
@@ -0,0 +1,87 @@
+#ifndef _COMPUTER_CPP
+#define _COMPUTER_CPP
+
+#include <iostream>
+# include <cstring>
+# include "Computer.h"
+    
+void Computer::setStr(const int& x, const char* str) {
+    switch (x) {
```

Comment:
Although original as idea I wouldn't have this method. You are not really sparing any code but adding "magic" number as parameter. It is impossible to remember after time what is 2 for example. It's better to keep the logic in each setter.